samples/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -233,6 +243,45 @@ static bool event_handler(const struct event_header *eh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
+		if (is_config_event(eh)) {
+			struct config_event *event = cast_config_event(eh);
+
+			if (!config_event_forward_needed(event->id)) {
+				goto error;
+			}
+
+			if (!bt_gatt_hids_ready_check(&hidc)) {
+				LOG_WRN("Cannot forward, peer disconnected");
+				goto error;
+			}
+
+			struct bt_gatt_hids_c_rep_info *config_rep =
+				bt_gatt_hids_c_rep_find(&hidc,
+					BT_GATT_HIDS_C_REPORT_TYPE_FEATURE,
+					REPORT_ID_USER_CONFIG);
+			if (config_rep == NULL) {
+				LOG_ERR("Feature report not found");
+				goto error;
+			}
+
+			u8_t data[REPORT_SIZE_USER_CONFIG];
+
+			data[0] = event->id;
+			memcpy(&(data[1]), event->data, sizeof(event->data));
+
+			int err = bt_gatt_hids_c_rep_write(&hidc, config_rep,
+					hidc_write_cb, data, sizeof(data));
+			if (err) {
+				LOG_ERR("Writing report failed");
+				goto error;
+			}
+
+			error:
```

Comment:
Is this an error? Most of the paths to error: are not directly errors.

Reply:
This is a way to skip execution, not only in error case. Maybe just `return false` will be cleaner.

Reply:
I think return false will look more correct.